### ISSUE TITLE
Electron-prebuilt reports ABI version 49?

### ISSUE THREAD

**Issue by gschmottlach** [electron-userland/electron-prebuilt#172]
I'm having difficulty running a C++ addon module with Electron-prebuilt v1.3.1. The module is using the headers from Electron pre-built and it builds successfully, but when I run it I get the following error:

Error: Module version mismatch. Expected 49, got 48.

Node 6.X.Y are all at ABI version 48. Electron Prebuilt v1.2.5 is reporting 48. But version 1.3.1 of Electron-Prebuilt reports 49. I thought (perhaps mistakenly) that these Electron ABI versions should track the corresponding Node.js version it's built against. Electron-Prebuilt v1.3.1 reports that it is built against Node.js 6.3.0 and this version of Node is at 48. Where did ABI version 49 come from? Was this a mistake when Electron-Prebuilt 1.3.1 was built?

**Comment by bgreenlee** [comment]
Having the same issue

**Comment by rgbkrk** [comment]
Confirmed. Versions up to `v1.2.9` are A-OK. `v1.3.x` is reporting an ABI _from the future_.

**Comment by bgreenlee** [comment]
Seems to still be an issue in 1.3.2

**Comment by Manicqin** [comment]
I have the same issue, I asked about it via the slack channel and I was referred to

http://electron.atom.io/docs/tutorial/using-native-node-modules/

I don't have access to my machine so I can't check it, but please do try.

**Comment by Manicqin** [comment]
Hi I read the link but I can' see how it should help my case.

One thing I noticed is that node-gyp (targeted to v.6.3.1) declares that my node version is: 4.4.7 even though I use the latest pre-built, I'm guessing it uses the global node instead the node distributed with electron.

How can I compile against the node that is shipped with pre-built instead of the global?
Is that even the problem?

Thanks.

**Comment by rgbkrk** [comment]
@Manicqin for your issue, you can make the compile use the local electron by using [`electron-rebuild`](https://github.com/electron/electron-rebuild)

**Comment by maxogden** [comment]
upstream issue https://github.com/electron/electron/issues/5851

**Comment by xdvarpunen** [comment]
Happening still, what's the fix?

**Comment by rgbkrk** [comment]
@xdvarpunen we wait on https://github.com/electron/electron/issues/5851. It's likely to happen on the next node 6 release, since v8 5.1 just landed on the node 6.x branch: https://github.com/nodejs/node/compare/92ecbc4...723fa96

**Comment by zaggino** [comment]
Any workaround until then except of reverting back to `1.2.8` ?

**Comment by rgbkrk** [comment]
I'm not sure, since the ABI is still 48 for node 6.5.

/cc @TheAlphaNerd @jasnell 🙏

**Comment by TheAlphaNerd** [comment]
we patched V8 5.1 to make it abi compatible. node 6.5.0 is still 48

**Comment by rgbkrk** [comment]
_Nice_ Thank you so much for chiming in. I'm guessing we need fixes in Electron land now.

**Comment by zeke** [comment]
The [electron-prebuilt repo is being retired](https://github.com/electron-userland/electron-prebuilt/pull/275) and its code has been moved into the electron/electron repo. For the sake of historical transparency, we will leave GitHub Issues enabled on this repository, but if you are still affected by the issue reported here, please [open a new issue on electron/electron](https://github.com/electron/electron/issues/new) repo and reference this issue from it so people can get the full context. The electron repository has a large and active contributor community, so your issue is more likely to get the attention it deserves there. Thanks!
